src/domain/file/services/file-service.ts

```diff
@@ -134,9 +134,10 @@ export class FileService {
         type: FileType,
     ): Promise<File> {
         if (type === FileType.MANUSCRIPT_SOURCE) {
-            const hasFile = await this.hasManuscriptFile(submissionId);
-            if (hasFile === true) {
-                throw new Error('Submission already has manuscript');
+            const manuscriptFile = await this.findManuscriptFile(submissionId);
+            // we have a file so delete it
+            if (manuscriptFile !== null) {
+                await this.deleteManuscript(manuscriptFile.id, submissionId);
```

Comment:
should we try-catch this call?

Reply:
Can do. Personally, if I was a user, I would prefer this process to halt if can't delete my files. It's possible I've uploaded the a file by accident and wish to ensure its deleted e.g. family photos, draft document, etc etc.

I generally edge on the side of caution. The delete can fail for two reasons - DB issue or S3, if the DB is failing to delete... we should know. If S3 is failing... we should know.

Reply:
I was thinking more about UX ... if it is an S3 problem we could carry on and successfully replace the manuscript - even though the old file still remains in S3 - as opposed to raising a visible error for the user? /cc @chugginselifesciences

Reply:
Maybe this is change of behaviour for the "deleteManuscript" function - as opposed to this one?

Reply:
I too was thinking about the UX. My personal view, if I was using a system then I generally like to know that when I delete my data then it's deleted as opposed potentially lingering around on S3 without my knowledge or consent. If a system fails to delete then it should inform me. The likely error rate is probably close to 0. S3 has an excellent uptime https://aws.amazon.com/s3/sla/

Reply:
Obviously, data deletion is not just a UX problem but also a a concern for data protection.

Reply:
> I too was thinking about the UX. My personal view, if I was using a system then I generally like to know that when I delete my data then it's deleted as opposed potentially lingering around on S3 without my knowledge or consent. If a system fails to delete then it should inform me. The likely error rate is probably close to 0. S3 has an excellent uptime https://aws.amazon.com/s3/sla/

In truth it would more likely be the network.